src/s_user.c

```diff
@@ -854,7 +854,7 @@ register_user(aClient *cptr, aClient *sptr, char *nick, char *username,
                                : "Too many connections from your site");
         }
 
-        if(!(ban = check_userbanned(sptr, UBAN_IP|UBAN_CIDR4, UBAN_WILDUSER)))
+        if(!(ban = check_userbanned(sptr, UBAN_IP|UBAN_CIDR4, 0)))
```

Comment:
Why is removing UBAN_WILDUSER necessary?

Reply:
If you look in check_userbanned, with the flag, during my tests, only bans on *@host instead of both and *user@host are being matched.